envs/monkey_zoo/blackbox/analyzers/mimikatz_analyzer.py

```diff
@@ -0,0 +1,65 @@
+from typing import List
+
+from common.credentials import Credentials
+from common.credentials.encoding import get_plaintext
+from envs.monkey_zoo.blackbox.analyzers.analyzer import Analyzer
+from envs.monkey_zoo.blackbox.analyzers.analyzer_log import AnalyzerLog
+from envs.monkey_zoo.blackbox.island_client.monkey_island_client import MonkeyIslandClient
+
+
+class MimikatzAnalyzer(Analyzer):
+    def __init__(
+        self, island_client: MonkeyIslandClient, expected_stolen_credentials: List[Credentials]
+    ):
+        self.island_client = island_client
+        self.expected_stolen_credentials = expected_stolen_credentials
+
+        self.log = AnalyzerLog(self.__class__.__name__)
+
+    def analyze_test_results(self) -> bool:
+        self.log.clear()
+
+        credential_stealing_success = self._analyze_credential_stealing()
+
+        return credential_stealing_success
+
+    def _analyze_credential_stealing(self) -> bool:
+        stolen_credentials = self.island_client.get_stolen_credentials()
+        plaintext_stolen_credentials = MimikatzAnalyzer._get_plaintext_stolen_credentials(
+            stolen_credentials
+        )
+
+        return self._all_expected_credentials_stolen(plaintext_stolen_credentials)
+
+    @staticmethod
+    def _get_plaintext_stolen_credentials(stolen_credentials: List[Credentials]) -> List[str]:
+        plaintext_stolen_credentials = set()
+
+        for credentials in stolen_credentials:
+            if credentials.identity:
+                plaintext_stolen_credentials.add(credentials.identity.username)
+            if credentials.secret:
+                # example of what the variables here look like:
+                #     credentials.secret = Password(password=SecretStr("abc")))
+                #     secret_tuple = ('password', SecretStr("abc"))
+                for secret_tuple in credentials.secret:
+                    plaintext_stolen_credentials.add(get_plaintext(secret_tuple[1]))
+
+        return list(plaintext_stolen_credentials)
+
```

Comment:
I don't know that this is necessary, I believe `Credentials` objects can be compared with the `==` operator. If we pass a `List[Credentials]` into the constructor, we should be able to just assert that
```python
len(stolen_credentials) == `len(expected_stolen_credentials`
for esc in expected_stolen_credentials:
    assert esc in stolen_credentials
```

Reply:
It'll make debugging easier. Any missing credentials are logged and doing it this way ensures that the plaintext credentials are logged. I suppose we could move this logic to where the missing credentials are actually logged but I don't think it makes much of a difference.

Reply:
See https://github.com/guardicore/monkey/pull/3276/files#r1182443171

Reply:
Moreover, credential pairs with our system now have meaning. That is, usernames and passwords that are stolen together are kept together so that brute forcers can try those first. The test really isn't testing that aspect if everything is broken apart into a list of strings with no associated types.